app.py

```diff
@@ -398,6 +465,137 @@ def index():
         'index.html'
     )
 
+
+
+
+
+# Subscribing/unsubscribing user from club
+
+@app.route('/subscriptions', methods=['GET'])
+def subscriptions():    
+    Session, engine = dbconnect(db_options)
+    session = Session()
+
+    subscriptions = session.query(UserToClubMapping).all()
+    
+    ret_subs = []
+    for s in subscriptions:
+        ret_subs.append(
+            {
+                'id': s.id,
+                'user_id': s.user_id,
+                'club_id': s.club_id,
+                'created_on': s.created_on,
+            }
+        )
+        #users = map(lambda u: dict(u), users)
+    return jsonify(ret_subs)
+
+
+@app.route('/subscribe', methods=['POST'])
+def subscribe():
+    if request.mimetype != 'application/json':
+        raise Exception('Content-Type is not "application/json".')
+    j = request.get_json()
+    Session, engine = dbconnect(db_options)
+    session = Session()
+    link = UserToClubMapping(
+        user_id=j.get('user_id'),
+        club_id=j.get('club_id'),
+    )
+
+    session.add(link)
+    session.flush()
+    session.commit()
+    return jsonify(
+        {
+            'user_id': link.user_id,
+            'club_id': link.club_id,
+        }
+    )
+
+@app.route('/unsubscribe', methods=['DELETE'])
+def unsubscribe():
+    if request.mimetype != 'application/json':
+        raise Exception('Content-Type is not "application/json".')
+    j = request.get_json()
+    Session, engine = dbconnect(db_options)
+    session = Session()
+    userID = j.get('user_id')
+    clubID = j.get('club_id')
+    # comma separated list of bool expr are AND
+    matchingLink = session.query(UserToClubMapping).filter(UserToClubMapping.user_id == userID, UserToClubMapping.club_id == clubID).one_or_none()
+    if matchingLink:
+        session.delete(matchingLink)
+        session.flush()
+        session.commit()
+        return jsonify(
+            {
+                'user_id': matchingLink.user_id,
+                'club_id': matchingLink.club_id,
+            }
+        )
+    else:
+        response = jsonify( {'message': 'Subscription between user ID and club ID not found'} )
+        response.status_code = 404
+        return response
+
+
+
+
```

Comment:
python best practice is two spaces between functions, one space between functions that are in the same class